images/krte/wrapper.sh

```diff
@@ -47,11 +47,15 @@ cleanup(){
 
 early_exit_handler() {
   >&2 echo "wrapper.sh] [EARLY EXIT] Interrupted, entering handler ..."
-  if [ -n "${WRAPPED_COMMAND_PID:-}" ]; then
+  # if we got here before the wrapped command exited, skip, otherwise signal and wait
+  if [ -n "${WRAPPED_COMMAND_PID:-}" && -z "${EXIT_VALUE:-}" ]; then
```

Comment:
```suggestion
  if [ -n "${WRAPPED_COMMAND_PID:-}" ] && [ -z "${EXIT_VALUE:-}" ]; then
```

https://tldp.org/LDP/abs/html/ops.html
> Example 8-3. Compound Condition Tests Using && and ||

Reply:
```sh
$ if [ -n "${WRAPPED_COMMAND_PID:-}" && -z "${EXIT_VALUE:-}" ]; then echo "si" ; fi
-bash: [: missing `]'
$ if [ -n "${WRAPPED_COMMAND_PID:-}" ] && [ -z "${EXIT_VALUE:-}" ]; then echo "si" ; fi
si
```